src/lib/project.js

```diff
@@ -308,9 +308,16 @@ async function scaffoldNext() {
   sh.rm('-rf', path.join('ui', '.git')); // Remove NextJS' .git; we will init .git in our monorepo's root.
   // Read in the NextJS config file and add the middleware.
   const nextConfig = fs.readFileSync(path.join('ui', 'next.config.js'), 'utf8');
-  const newNextConfig = nextConfig.replace(
+  let newNextConfig = nextConfig.replace(
     /^}(.*?)$/gm, // Search for the last '}' in the file.
     `
+  webpack(config) {
+    config.resolve.alias = {
+      ...config.resolve.alias,
+      snarkyjs: require('path').resolve('./node_modules/snarkyjs'),
```

Comment:
I'm curious why this is required

Reply:
This ensures that the smart contract's snarkyjs and the ui's snarkyjs are the same - otherwise they get imported multiple times. See [here for more](https://blog.maximeheckel.com/posts/duplicate-dependencies-npm-link/), in the context of fixing this for duplicate instances of react

Reply:
Peer dependancies appeared to fix this issue for contracts imported via npm but not via npm link

Reply:
thanks @es92 this link is a nice resource! another potential trick with npm link is mentioned here (at the bottom): https://reactjs.org/warnings/invalid-hook-call-warning.html#duplicate-react